src/Lecture3.hs

```diff
@@ -92,7 +95,10 @@ weekday to the second.
 >>> daysTo Friday Wednesday
 5
 -}
-daysTo = error "TODO"
+daysTo :: (Ord a, Enum a, Bounded a) => a -> a -> Int
+daysTo x y
+    | x <= y = fromEnum y - fromEnum x
+    | otherwise = fromEnum (maxBound `asTypeOf` x) - fromEnum x + fromEnum y + 1
```

Comment:
But here you do need to use `asTypeOf` (or something similar) because if you just do `fromEnum maxBound`, Haskell doesn't know which `maxBound` you mean 👍🏻 

You also implemented a very generic function here and that's great 👏🏻